okhttp/src/jvmMain/kotlin/okhttp3/Route.kt

```diff
@@ -87,5 +86,42 @@ class Route(
     return result
   }
 
-  override fun toString(): String = "Route{$socketAddress}"
+  /**
+   * Returns a string with the URL hostname, socket IP address, and socket port, like one of these:
+   *
+   *  * `example.com:80 at 1.2.3.4:8888`
+   *  * `example.com:443 via proxy [::1]:8888`
+   *
+   * This omits duplicate information when possible.
+   */
+  override fun toString(): String {
```

Comment:
Do we use this at least once, maybe more. Is it worth making this lazy and avoid doing twice?!

Reply:
We're currently using it zero times. With happy eyeballs we'll use it once to name a task that ultimately names a thread.

I don't think it benefits from caching in our current use cases. Caching stuff like this is most useful if we use the string form programmatically, like if we were using the string as a HashMap key.

Also, I reserve the right to change my mind on this if YourKit tells me to!